packages/docusaurus/src/webpack/utils.ts

```diff
@@ -175,6 +176,33 @@ export function applyConfigureWebpack(
   return config;
 }
 
+export function applyConfigurePostCss(
```

Comment:
was wondering if we need a new lifecycle for this case?

Maybe that should be part of configureWebpack? (as the post css loader is part of the webpack config?)

No strong opinions for now, just thinking out loud

Reply:
Of course, PostCSS plugins can be added via `configureWebpack` hook, but for this we need to duplicate a lot Docusaurus core code related to webpack, for example from this [comment](https://github.com/facebook/docusaurus/issues/2961#issuecomment-735355912):

![image](https://user-images.githubusercontent.com/4408379/107271177-eb2c0c00-6a5c-11eb-957d-9312a8f0effc.png)

(Since this hook should return a valid webpack config, which will be merged with the default config).

Thus, `configurePostCss` is a handy and useful shortcut that you can use to quickly add the required PostCSS plugins. Or do you know a better way to achieve this?

Reply:
Thanks @lex111 , understand, and it will be useful for the upcoming tailwind theme too

What I see is that Gatsby does something similar with a plugin: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-postcss/src/gatsby-node.js

And next does not really have a plugin system, but allow a local `postcss.config.js` and will support tailwind postcss enhancements in core soon: https://github.com/vercel/next.js/discussions/20030.

That does not look viable for our usecase so the most convenient is indeed a new lifecycle as themes can enhance the existing config.

Going to finish your implementation and merge, I suspect there is a little bug for which I want to have a test, and will also try to make the loader/option lookup more robust